jobs/resend_invitation_email/worker.go

```diff
@@ -64,6 +67,76 @@ func (rseworker *ResendInvitationEmailWorker) JobChannel() chan<- model.Job {
 	return rseworker.jobs
 }
 
+func (rseworker *ResendInvitationEmailWorker) DoJob(job *model.Job) {
+	resendInviteEmailIntervalFlag := rseworker.App.Config().FeatureFlags.ResendInviteEmailInterval
+
+	switch resendInviteEmailIntervalFlag {
+	case "48":
+		rseworker.DoJob_24_48(job)
+	case "72":
+		rseworker.DoJob_24_72(job)
+	default:
+		rseworker.DoJob_24(job)
+	}
+}
+
+func (rseworker *ResendInvitationEmailWorker) DoJob_24(job *model.Job) {
+	elapsedTimeSinceSchedule, DurationInMillis_24, _, _ := rseworker.GetDurations(job)
+	if elapsedTimeSinceSchedule > DurationInMillis_24 {
+		rseworker.ResendEmails(job)
+		rseworker.TearDown(job)
+	}
+}
+
+func (rseworker *ResendInvitationEmailWorker) DoJob_24_48(job *model.Job) {
+	elapsedTimeSinceSchedule, DurationInMillis_24, DurationInMillis_48, _ := rseworker.GetDurations(job)
+	systemValue, sysValErr := rseworker.App.Srv().Store.System().GetByName(model.NumberOfInviteEmailsSent)
+	if sysValErr != nil {
+		if _, ok := sysValErr.(*store.ErrNotFound); !ok {
+			mlog.Error("An error occurred while getting NUMBER_OF_INVITE_EMAILS_SENT from system store", mlog.String("worker", rseworker.name), mlog.Err(sysValErr))
+			return
+		}
+	}
+	if (elapsedTimeSinceSchedule > DurationInMillis_24) && (elapsedTimeSinceSchedule < DurationInMillis_48) && (systemValue.Value == "0") {
```

Comment:
Are we checking somewhere if the user already joined? Why do we check if it was previously sent?

Reply:
Yes. The function `ResendEmails` calls `removeAlreadyJoined` to take care of that. The comparison `systemValue.Value == "0"` is for the job to not continously run the first branch of the if statement. I do this because the times are not exact. Without it, between `elapsedTimeSinceScshedule` and `DurationInMillis_48`, the first branch of the if statement can run several times. So, we want to mark it done.

Reply:
Makes sense, thanks for clarifying!